app/services/no_bp_measure_service.rb

```diff
@@ -0,0 +1,88 @@
+class NoBPMeasureService
+  CACHE_VERSION = 1
+
+  def initialize(region, periods:, type: :missed_visits)
+    @region = region
+    @periods = periods
+    @facilities = region.facilities.to_a
+    @type = type
+  end
+
+  attr_reader :facilities
+  attr_reader :periods
+  attr_reader :region
+  attr_reader :type
+
+  def call
+    Rails.cache.fetch(cache_key, version: cache_version, expires_in: 7.days, force: force_cache?) do
+      periods.each_with_object({}) do |period, result|
+        result[period] = missed_visits_for(period)
+      end
+    end
+  end
+
+  def missed_visits_for(period)
+    if type == :missed_visits
+      visit_start_range = period.advance(years: -1).start_date
+      visit_end_range = period.advance(months: -3).start_date
+    else
+      visit_start_range = "-infinity"
```

Comment:
This is [a valid Postgres special data/time](https://www.postgresql.org/docs/10/datatype-datetime.html#DATATYPE-DATETIME-SPECIAL-TABLE).